watertap/unit_models/zero_order/tests/test_zo_documentation.py

```diff
@@ -0,0 +1,64 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+Tests that documentation exists for all zero-order unit models
+"""
+import pytest
+import os
+
+from pyomo.common.fileutils import this_file_dir
+
+this_file_dir = this_file_dir()
+
+doc_path = os.path.join(
+    this_file_dir,
+    "..",
+    "..",
+    "..",
+    "..",
+    "docs",
+)
+
+rst_path = os.path.join(
+    doc_path,
+    "technical_reference",
+    "unit_models",
+    "zero_order_unit_models",
+)
+
+py_path = os.path.join(this_file_dir, "..")
+
+
+def generate_model_list(path, extension, exclude):
+    file_list = []
+    for file in os.listdir(path):
+        filename = os.fsdecode(file)
+        if filename.endswith(extension) and filename not in exclude:
+            file_path = os.path.join(
+                path,
+                file,
+            )
+            assert os.path.isfile(file_path)
+            file_list.append(filename[: -len(extension)])
+    return file_list
+
+
+@pytest.mark.integration
+@pytest.mark.skipif(not os.path.exists(doc_path), reason="No docs directory")
```

Comment:
I think the reason why the tests are failing is that `doc_path` exists (so the test doesn't get skipped), but `rst_path` doesn't (so the test fails). Therefore, it might work with this change:

```suggestion
@pytest.mark.skipif(not os.path.exists(rst_path), reason="Directory with .rst source files for ZO models docs not found")
```

Reply:
On second thought, reading https://github.com/watertap-org/watertap/pull/877#issuecomment-1371266240 and https://github.com/watertap-org/watertap/pull/877#issuecomment-1371293292, I don't think this would work: the directory would exist, but without any `.rst` file.

Therefore, testing for the presence of at least one `.rst` file should work:

```suggestion
doc_path = Path(doc_path).resolve()

...

@pytest.mark.skipif(not os.path.exists(doc_path / "index.rst"), reason="No RST files found")
```

However, this starts to look a bit too brittle for my tastes...